pkg/apic/provisioning/requeststatus.go

```diff
@@ -0,0 +1,69 @@
+package provisioning
+
+import "fmt"
+
+const statusSetError = "can not set status as it's already been set"
+
+// RequestStatus - holds info about the status of the request
+type RequestStatus struct {
+	status  status
+	message string
+}
```

Comment:
May be we can include the reference here, the caller of the provisioning interface methods can use it and associate it back to the resource (ManagedApplication, AccessRequest, Credential).